### PR TITLE
Updated Guitar for Swift 4.0

### PR DESCRIPTION
**Pull request by ArtSabintsev**
### Pull Request Description

Updated Guitar's commit hash and version as it now supports Swift 4.0.

### Acceptance Criteria

To be accepted into the Swift source compatibility test suite, a project must:

- [x] be an *Xcode* or *swift package manager* project
- [x] support building on either Linux or macOS
- [x] target Linux, macOS, or iOS/tvOS/watchOS device
- [ x be contained in a publicly accessible git repository
- [x] maintain a project branch that builds against Swift 3.0 compatibility mode
      and passes any unit tests
- [x] have maintainers who will commit to resolve issues in a timely manner
- [x] be compatible with the latest GM/Beta versions of *Xcode* and *swiftpm*
- [x] add value not already included in the suite
- [x] be licensed with one of the following permissive licenses:
	* BSD
	* MIT
	* Apache License, version 2.0
	* Eclipse Public License
	* Mozilla Public License (MPL) 1.1
	* MPL 2.0
	* CDDL
- [x] pass ./check script run

Ensure project meets all listed requirements before submitting a pull request.

### CODE REVIEW

**Comment by ArtSabintsev** [comment]
Getting some issues with `check` script:

```
$ ./check script run
usage: check [-h] [--swiftc PATH] project
check: error: unrecognized arguments: run
```

**Comment by lplarson** [comment]
@swift-ci test

**Comment by lplarson** [comment]
Please use the check script like so: ./check project-path-field

**Comment by ArtSabintsev** [comment]
Here's what I get:

```
Arthur: ~/Documents/oss/swift-source-compat-suite 
(master)$ ./check Guitar
** CHECK **
--- Finding Guitar Swift version compatibility ---
--- Project configured to be compatible with Swift 4.0 ---
--- Checking Guitar platform compatibility with Darwin ---
--- Platform compatibility check succeeded ---
--- Locating swiftc executable ---
$ xcrun -f swiftc
--- Checking installed Swift version ---
$ /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc --version
error: Project configured for unsupported platform or Swift version
Traceback (most recent call last):
  File "./check", line 153, in <module>
    sys.exit(main())
  File "./check", line 150, in main
    return check(args.project, swift_branch, args.swiftc, compatibility_version)
  File "./check", line 123, in check
    if not is_correct_swift_version(swiftc, compatibility_version):
  File "./check", line 98, in is_correct_swift_version
    expected_swift_version = supported_configs[platform.system()][compatibility_version]['version']
KeyError: u'4.0'
```

**Comment by ArtSabintsev** [comment]
Follow up:

```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc --version
Apple Swift version 4.0 (swiftlang-900.0.65 clang-900.0.37)
Target: x86_64-apple-macosx10.9
```

What version is it expecting?

**Comment by ArtSabintsev** [comment]
Well, looks like it passed!

File: projects.json [hunk 1]
```diff
@@ -412,8 +412,8 @@
     "path": "Guitar",
     "branch": "master",
     "compatibility": {
-      "3.1": {
```
Comment (lplarson) [inline projects.json]: I'd like to maintain compatibility testing for 3.1. Could you please add it back, so we have both 3.1 and 4.0 hashes?

**Comment by lplarson** [comment]
Don't worry about the ./check script issue. 4.0 support hasn't been added to it yet.

File: projects.json (hunk 1 above)
Comment (ArtSabintsev) [reply projects.json]: sure - will do it soon.

**Comment by ArtSabintsev** [comment]
Alright, should be good to go now, I hope.

**Comment by lplarson** [comment]
@swift-ci test

**Comment by lplarson** [comment]
@swift-ci test

**Comment by ArtSabintsev** [comment]
Looks like we're golden @lplarson

**Comment by lplarson** [comment]
Looks good. Thanks!

**Pull request merged**